pkg/cli/state/vmclarity.go

```diff
@@ -427,11 +427,17 @@ func appendEffectiveScanConfigAnnotation(annotations *models.Annotations, config
 			Value: utils.PointerTo(string(configJSON)),
 		},
 	}
-
 	if annotations == nil {
 		return &effectiveScanConfigAnnotations, nil
 	}
-	newAnnotations := *annotations
+	var newAnnotations models.Annotations
+	for _, annotation := range *annotations {
+		// If effective scan config annotation exists we will overwrite it.
+		if *annotation.Key == effectiveScanConfigAnnotationKey {
+			continue
+		}
+		newAnnotations = append(newAnnotations, annotation)
+	}
 	newAnnotations = append(newAnnotations, effectiveScanConfigAnnotations...)
```

Comment:
I think we can clean this up like:
```
func appendEffectiveScanConfigAnnotation(annotations *models.Annotations, config *families.Config) (*models.Annotations, error) {
	var newAnnotations models.Annotations
	
	if annotatations != nil {
		// Add all odd annotations accept for the effective scan config one if they exist.
		for _, annotation := range *annotations {
			if *annotation.Key == effectiveScanConfigAnnotationKey {
				continue
			}
			newAnnotations = append(newAnnotations, annotation)
		}
	}

    // Add the new effective scan config annotation
	configJSON, err := json.Marshal(config)
	if err != nil {
		return nil, fmt.Errorf("failed to marshal effective families config: %w", err)
	}
	newAnnotations = append(newAnnotations, {
			Key:   utils.PointerTo(effectiveScanConfigAnnotationKey),
			Value: utils.PointerTo(string(configJSON)),
	})
	return &newAnnotations, nil
}
```

Reply:
https://github.com/openclarity/vmclarity/pull/739/commits/89fb81ae0783eb6935126ab5c5838945d4e26980